### PR TITLE
Support the `|defer` flag in the Template::generateScriptTag() method

### LINKED ISSUE #692: add defered scripts not possible
$GLOBALS['TL_JAVASCRIPT'][] = 'script.js|async';
$GLOBALS['TL_JAVASCRIPT'][] = 'script.js|static';

Works fine, but:

$GLOBALS['TL_JAVASCRIPT'][] = 'script.js|defer';
is not supported?

If so, I think it should also be supported.

### PR DESCRIPTION
**Pull request by Wusch**
Fixes #692

### CODE REVIEW

**Review by Toflar — APPROVED** [review]
<3

File: core-bundle/src/Resources/contao/library/Contao/Controller.php [hunk 1]
```diff
@@ -968,6 +980,24 @@ public static function replaceDynamicScriptTags($strBuffer)
...
+				{
+					$arrReversed = array_reverse($objCombinerDefer->getFileUrls());
+
+					foreach ($arrReversed as $strUrl)
+					{
+						$options = StringUtil::resolveFlaggedUrl($strUrl);
+						$strScripts = Template::generateScriptTag($strUrl, false, $options->mtime, null, null, null, true) . $strScripts;
+					}
+				}
+			}
```
Comment (fritzmg) [inline core-bundle/src/Resources/contao/library/Contao/Controller.php:1000]: Shouldn't this come directly after the `if ($objCombinerAsync->hasEntries())` block, within the same `if (!empty($GLOBALS['TL_JAVASCRIPT']) && \is_array($GLOBALS['TL_JAVASCRIPT']))` block?

Comment (Wusch) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1000]: It is. Github displays this a bit weird.

Comment (fritzmg) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1000]: Sorry, should have checked in the orignal source file :)

**Review by fritzmg — APPROVED** [review]

**Review by aschempp — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Wusch.

**Pull request merged**